pages/source/documentation/index.md

```diff
@@ -86,8 +87,50 @@ Dynflow has been developed to be able to support orchestration of services in th
 
 ### Action anatomy
 
-Each action can be viewed as a function. It's planned for execution with 
-input, then it's executed producing output quite possibly having side-effects. 
+When action is triggered, Dynflow executes plan method on this action, which
+is responsible to build the plan. It builds the plan by calling plan_action
+and plan_self methods, effectively listing actions that should be run as 
+a part of this plan. In other words, it compiles a list of actions on which 
+method run will be called. Also it's responsible for giving these actions 
```

Comment:
Does plan method deals with finalize anyhow? I thought it's isolated and is triggered only if it's defined for the action.